src/dsp_tools/commands/validate_data/reformat_validaton_result.py

```diff
@@ -205,31 +218,65 @@ def _reformat_one_with_detail(val_result: ResultWithDetail) -> InputProblem | Un
     subject_id = _reformat_data_iri(str(val_result.res_iri))
     prop_name = _reformat_onto_iri(str(val_result.property))
     res_type = _reformat_onto_iri(str(val_result.res_class))
-    match val_result.detail_bn_component:
+    match val_result.detail.component:
         case SH.PatternConstraintComponent:
-            val: str | None = val_result.value
+            val: str | None = val_result.detail.value
             if val and not regex.search(r"\S+", val):
                 val = None
             return ContentRegexViolation(
                 res_id=subject_id,
                 res_type=res_type,
                 prop_name=prop_name,
-                expected_format=val_result.results_message,
+                expected_format=val_result.detail.results_message,
                 actual_content=val,
             )
-        case SH.ClassConstraintComponent | SH.MinCountConstraintComponent:
-            actual_type = _reformat_onto_iri(str(val_result.value_type)).replace("knora-api:", "")
+        case SH.ClassConstraintComponent:
+            return _reformat_detail_class_constraint_component(val_result)
+        case SH.MinCountConstraintComponent:
+            actual_type = _reformat_onto_iri(str(val_result.detail.value_type)).replace("knora-api:", "")
             return ValueTypeViolation(
                 res_id=subject_id,
                 res_type=res_type,
                 prop_name=prop_name,
                 actual_type=actual_type,
-                expected_type=val_result.results_message,
+                expected_type=val_result.detail.results_message,
             )
         case _:
             return UnexpectedComponent(str(val_result.source_constraint_component))
 
 
+def _reformat_detail_class_constraint_component(val_result: ResultWithDetail) -> InputProblem:
+    subject_id = _reformat_data_iri(str(val_result.res_iri))
+    prop_name = _reformat_onto_iri(str(val_result.property))
+    res_type = _reformat_onto_iri(str(val_result.res_class))
+    actual_type = _reformat_onto_iri(str(val_result.detail.value_type)).replace("knora-api:", "")
+    match val_result.detail.result_path:
+        case API_SHAPES.linkValueHasTargetID:
+            value = _reformat_data_iri(str(val_result.detail.value))
+            if val_result.detail.results_message == "Resource":
```

Comment:
This is not a very good solution.
In my next PR I am planning on completely refactoring the queries for result details. The requirements for the different validation reports are too different to query all the information and do triage then. It is not feasible with a large graph. But I want to implement these changes in a separate PR that is not also a feature.